access-checkout/src/main/java/com/worldpay/access/checkout/ui/AccessEditText.kt

```diff
@@ -18,47 +20,73 @@ class AccessEditText @JvmOverloads constructor(
     defStyle: Int = 0
 ) : LinearLayout(context, attrs, defStyle) {
     private var mCustomHint: String? = null
-    private lateinit var editText: EditText
+    private var editText: EditText? = null
 
     // with the internal access modifier, this property is internal to the Access Checkout SDK JAR file
     // (or access-checkout Gradle project when adding the dependency as a project dependency)
     // and cannot be accessed outside of that module, e.g. cannot be accessed by a merchant's app
-    internal val text: String get() = editText.text.toString()
+    internal val text: String get() = editText!!.text.toString()
 
-    val selectionEnd: Int get() = editText.selectionEnd
+    val selectionEnd: Int get() = editText!!.selectionEnd
 
-    val editableText: Editable get() = editText.editableText
+    val selectionStart: Int get() = editText!!.selectionStart
 
-    val isCursorVisible: Boolean get() = editText.isCursorVisible
+    val editableText: Editable get() = editText!!.editableText
+
+    val isCursorVisible: Boolean get() = editText!!.isCursorVisible
+    val currentTextColor: Int get() = editText!!.currentTextColor
 
     var filters: Array<InputFilter>
         get() {
-            return editText.filters
+            return editText!!.filters
         }
         set(filters) {
-            editText.filters = filters
+            editText!!.filters = filters
         }
 
     var inputType: Int
         get() {
-            return editText.inputType
+            return editText!!.inputType
         }
         set(inputType) {
-            editText.inputType = inputType
+            editText!!.inputType = inputType
+        }
+
+    var keyListener: KeyListener
```

Comment:
This must be internal

Reply:
Reason is it could allow a merchant to listen to individual keys and capture the card number that way